tools/cellprofiler/starting_modules_metadata.xml

```diff
@@ -0,0 +1,96 @@
+                                    <option value="(?P&lt;field1>.*)-(?P&lt;field2>[a-zA-Z0-9]+)-(?P&lt;field3>[a-zA-Z0-9]+)-(?P&lt;field4>[a-zA-Z0-9]+)-(?P&lt;field5>[a-zA-Z0-9]+)-(?P&lt;field6>[a-zA-Z0-9]+)">field1-field2-field3-field4-field5-field6</option>
+                                    <option value="(?P&lt;field1>.*)_(?P&lt;field2>[a-zA-Z0-9]+)_(?P&lt;field3>[a-zA-Z0-9]+)_(?P&lt;field4>[a-zA-Z0-9]+)_(?P&lt;field5>[a-zA-Z0-9]+)_(?P&lt;field6>[a-zA-Z0-9]+)">field1_field2_field3_field4_field5_field6</option>
+                                    <option value="(?P&lt;field1>.*)__(?P&lt;field2>[a-zA-Z0-9]+)__(?P&lt;field3>[a-zA-Z0-9]+)__(?P&lt;field4>[a-zA-Z0-9]+)__(?P&lt;field5>[a-zA-Z0-9]+)__(?P&lt;field6>[a-zA-Z0-9]+)">field1__field2__field3__field4__field5__field6</option>
+                                </param>
+                            </when>
+                            <when value="Folder name">
+                                <param name="folder_name_regex" type="select" label="Select the pattern to extract metadata from the folder name" help="Folder names must comply with one of the patterns. For example, folder name exp1-channel_name matches the pattern field1-field2. If none of the patterns is suitable, please use other Galaxy tools to rename your folder first.">
+                                    <sanitizer sanitize="false"/>
+                                    <option value="(?P&lt;folderField1>.*)">field1</option>
+                                    <option value="(?P&lt;folderField1>.*)-(?P&lt;folderField2>[a-zA-Z0-9]+)">field1-field2</option>
+                                    <option value="(?P&lt;folderField1>.*)_(?P&lt;folderField2>[a-zA-Z0-9]+)">field1_field2</option>
+                                </param>
+                            </when>
+                        </conditional>
+                        <conditional name="con_metadata_extract_from">
+                            <param name="extract_metadata_from" type="select" label="Extract metadata from">
+                                <option value="All images">All images</option>
+                                <option value="Images matching a rule">Images matching a rule</option>
+                            </param>
+                            <when value="Images matching a rule">
+                                <expand macro="image_matching_rules_metadata" />
+                            </when>
+                            <when value="All images" />
+                        </conditional>
+                    </repeat>
+                    <param name="metadata_type" type="select" label="Metadata data type">      
+                        <option value="Text">Save all metadata items as text</option>
+                    </param>

```

Comment:
shouldn't we remove this param? "Choose for each" we cannot have, so there's no choice for "Metadata data type". I would just assume in the cppipe that the default value is "Text".

Reply:
ok..i will update it soon